cmd/parca-agent/main.go

```diff
@@ -364,9 +365,15 @@ func main() {
 	// Run group for target manager
 	{
 		ctx, cancel := context.WithCancel(ctx)
+		counter := promauto.With(reg).NewCounterVec(
```

Comment:
Let's think a little ahead and create a `metrics` struct here to collect similar metrics.